addons/survey/models/survey.py

```diff
@@ -0,0 +1,1066 @@
+            'name': "Start Survey",
+            'target': 'self',
+            'url': self.with_context(context).public_url + trail
+        }
+
+    @api.multi
+    def action_send_survey(self):
+        ''' Open a window to compose an email, pre-filled with the survey
+        message '''
+        self.ensure_one()
+        if not self.designed:
+            raise UserError(_('You cannot send an invitation for a survey that has no questions.'))
+
+        if self.stage_id.closed:
+            raise UserError(_("You cannot send invitations for closed surveys."))
+        template_id = self.env.ref('survey.email_template_survey', raise_if_not_found=False)
+        ctx = dict(self.env.context)
+        ctx.update({'default_model': 'survey.survey',
+                    'default_res_id': self.id,
+                    'default_survey_id': self.id,
+                    'default_use_template': bool(template_id),
+                    'default_template_id': template_id and template_id.id or False,
+                    'default_composition_mode': 'comment'}
+                   )
+        return {
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'res_model': 'survey.mail.compose.message',
+            'target': 'new',
+            'context': ctx,
+        }
+
+    @api.multi
+    def action_print_survey(self):
+        ''' Open the website page with the survey printable view '''
+        self.ensure_one()
+        trail = ""
+        context = dict(self.env.context or {}, relative_url=True)
+        if 'survey_token' in context:
+            trail = "/" + context['survey_token']
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "Print Survey",
+            'target': 'self',
+            'url': self.with_context(context).print_url + trail
+        }
+
+    @api.multi
+    def action_result_survey(self):
+        ''' Open the website page with the survey results view '''
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "Results of the Survey",
+            'target': 'self',
+            'url': self.with_context(relative_url=True).result_url
+        }
+
+    @api.multi
+    def action_test_survey(self):
+        ''' Open the website page with the survey form into test mode'''
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "Results of the Survey",
+            'target': 'self',
+            'url': self.with_context(relative_url=True).public_url + "/phantom"
+        }
+
+    # Public methods #
+
+    def copy_data(self, cr, uid, id, default=None, context=None):
+        current_rec = self.read(cr, uid, id, fields=['title'], context=context)
+        title = _("%s (copy)") % (current_rec.get('title'))
+        default = dict(default or {}, title=title)
+        return super(Survey, self).copy_data(cr, uid, id, default, context=context)
```

Comment:
After checking this issue I found that copy_data or models.py is not decorated explicitly and its signature is:
``` def copy_data(self, cr, uid, id, default=None, context=None): ```
So by default it will be decorated with cr_uid_id_context(using guess).

If we convert copy_data method in our model(survey.survey), the call will be in new_api, and as copy_data of BaseModel is decorated with cr_uid_id_context, it will call copy_data method in loop and result will list of dictionary.

Workaround should be override copy method(if it is really not necessary to override copy_data method).

And yes copy_data method should be decorated explicitly in BaseModel and should be converted to new style, as we override this method at many places in addons.
@rco-odoo what do you think ?